lib/handlers.js

```diff
@@ -0,0 +1,54 @@
+
+module.exports = (self) => {
+  return {
+    'apostrophe:modulesRegistered': {
+      setContextOperations() {
+        const excludedExportTypes = [];
+        const excludedImportTypes = [];
+        for (const mod of Object.values(self.apos.modules)) {
+          if (!self.apos.instanceOf(mod, '@apostrophecms/doc-type')) {
+            continue;
+          }
+
+          if (mod.options.importExport?.export === false) {
+            excludedExportTypes.push({
+              type: {
+                $ne: mod.__meta.name
+              }
+            });
+          }
+
+          if (!mod.options.singleton || mod.options.importExport?.import === false) {
```

Comment:
I hope that `option.singletonAuto` will also be picked up, because `modulesRegistered` should be executed after all `init` operations, right?

Reply:
Yes at this stage a module with `singletonAuto` have seen its options updated to also have `singleton` true.
tested!